# CASE 01 -  Sales Force Training
Perusahaan X ingin meningkatkan penjualan mereka. Dari data penjualan 
sebelumnya menunjukkan bahwa penjualan rata-rata yaitu $100 per 
transaksi. Setelah melakukan training kepada pekerja sales, data 
penjualan terbaru (yang diambil dari 25 sampel pekerja sales) tersimpan 
Lakukan beberapa langkah berikut:
1. Buatlah DataFrame 1 kolom dengan 25 baris bernama ‘TransactionAmount’
2. Buatlah analisis measure of central tendency dari data ‘TransactionAmount’
3. Lakukan juga analisis measure of variability dari data ‘TransactionAmount’
4. Tentukan H0 & H1 untuk persiapan dilakukan T-Test
5. Lakukan proses T-Test pada data rata-rata jumlah transaksi (dengan alpha = 5%)
6. Buatlah kesimpulan dari hasil pengujian sales force training!

In [1]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [3]:
#Create DataFrame
transaction_data = [100, 150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 50,
                   100, 100, 140, 90, 150, 50, 90, 120, 100, 110, 75, 65]
df = pd.DataFrame({'TransactionAmount' : transaction_data})
print(df)

    TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65


In [5]:
#Central Tendency
import statistics
print(statistics.mean(transaction_data))
print(statistics.median(transaction_data))

102.6
100


#### Dari analisis central tendency yang meliputi mean, median dari data 'TransactionAmount':
1. Nilai Mean berbeda dari nilai sebelumnya ($100) yaitu 102.6, ini bisa menandakan peningkatan efektivitas pelatihan

2. Nilai Median sama dari nilai sebelumnya ($100) hal ini bisa memberikan gambaran yang lebih stabil tentang perubahan dalam distribusi data.

In [7]:
#Variability
import numpy as np
print(np.var(transaction_data))
print(np.std(transaction_data))
print(np.max(transaction_data) - np.min(transaction_data))
print(np.quantile(transaction_data, 0.25))
print(np.quantile(transaction_data, 0.5))
print(np.quantile(transaction_data, 0.75))

972.2399999999999
31.18076330047101
100
85.0
100.0
120.0


#### Analisis variability dari data 'TransactionAmount':
Variance (Variansi): Hasil dari np.var(transaction_data) adalah sekitar 972.24. Ini menunjukkan seberapa jauh nilai-nilai dalam data 'TransactionAmount' menyebar dari rata-rata. Variansi yang tinggi menandakan adanya variasi yang besar di antara nilai-nilai tersebut.

Standard Deviation (Deviasi Standar): Hasil dari np.std(transaction_data) adalah sekitar 31.18. Deviasi standar merupakan akar kuadrat dari varians, mengukur seberapa jauh nilai-nilai dalam dataset tersebar dari nilai rata-rata. Semakin besar nilai deviasi standar, semakin besar variasi data dari nilai rata-rata.

Range (Rentang): Rentang antara nilai tertinggi dan terendah dari data adalah 100. Ini menunjukkan perbedaan antara nilai transaksi tertinggi dan terendah dalam dataset.

Kuartil (Quartile): Quartile menyajikan titik-titik yang membagi data menjadi empat bagian yang sama besar. Di sini, kita melihat kuartil pertama (Q1) sebesar 85.0, kuartil kedua (median) sebesar 100.0, dan kuartil ketiga (Q3) sebesar 120.0. Median menunjukkan nilai tengah dari data ketika diurutkan dari yang terkecil hingga yang terbesar.


#### Define H0 and H1
H0 : the average of training sales transactions amount = 100

H1 : the average of training sales transactions amount > 100

##### Define alpha = 0.05

In [9]:
#Calculate t statistics and P-value
#Perform one sample t-test

stats.ttest_1samp(a=transaction_data, popmean=100)

TtestResult(statistic=0.4085001556802841, pvalue=0.6865284813438117, df=24)

In [10]:
#statistic=0.4085001556802841, pvalue=0.6865284813438117

#### Kesimpulan dari hasil ini akan didasarkan pada nilai p-value yang diperoleh dalam analisis:

Karenailai p-value (0.6865) lebih besar dari alpha (0.05 atau 5%). Maka kita gagal menolak H0

Kesimpulanya : training pada sales tidak efektif dan rata-rata jumlah transaksi masih $100


In [11]:
#Conclude with t statistics and critical region
#Find the critical region

In [12]:
#H1: larger (right side)
stats.t.ppf (1-0.05, 19)

1.729132811521367

#### Kesimpulan :
    
Karena nilai critical region: t > 1.73
    
as the statistic is not in the critical region, then we fail to reject H0

training pada sales tidak efektif dan rata-rata jumlah transaksi masih $100
